ibis/file/client.py

```diff
@@ -0,0 +1,137 @@
+import ibis
+import ibis.expr.types as ir
+from ibis.pandas.core import execute, execute_with_scope  # noqa
+
+try:
+    import pathlib
+except ImportError:
+
+    # py2 compat
+    import pathlib2 as pathlib
+
+
+class FileClient(ibis.client.Client):
+
+    def __init__(self, root):
+        self.root = pathlib.Path(str(root))
+        self.dictionary = {}
+
+    def insert(self, path, expr, **kwargs):
+        raise NotImplementedError
+
+    def table(self, name, path):
+        raise NotImplementedError
+
+    def database(self, name=None, path=None):
+        if name is None:
+            return FileDatabase('root', self, path=path)
+
+        if name not in self.list_databases(path):
+            raise AttributeError(name)
+        if path is None:
+            path = self.root
+
+        new_name = "{}.{}".format(name, self.extension)
+        if (self.root / name).is_dir():
+            path /= name
+        elif not str(path).endswith(new_name):
+            path /= new_name
+
+        return FileDatabase(name, self, path=path)
+
+    def execute(self, expr, params=None, **kwargs):  # noqa
+        assert isinstance(expr, ir.Expr)
+        scope = kwargs.pop('scope', {})
+        return execute_with_scope(
```

Comment:
Just call `execute` here. `execute_with_scope` isn't for use by anyone or anything except `execute` itself.

Reply:
done